common/djangoapps/student/tests/test_email.py

```diff
@@ -150,6 +150,32 @@ def _assert_activation_email(self, subject, body_fragments):
         for fragment in body_fragments:
             self.assertIn(fragment, msg.body)
 
+    def test_do_not_send_email_and_do_activate(self):
+        """
+        Tests that when an inactive user logs-in using the social auth,
+        an activation email is not sent.
+        """
+        pipeline_partial = {
+            'kwargs': {
+                'social': {
+                    'uid': 'fake uid'
+                }
+            }
+        }
+        user = UserFactory(is_active=False)
+        Registration().register(user)
+        request = RequestFactory().get(settings.SOCIAL_AUTH_INACTIVE_USER_URL)
+        request.user = user
+        with patch('student.views.management.compose_and_send_activation_email') as email:
+            with patch('third_party_auth.provider.Registry.get_from_pipeline') as reg:
+                with patch('third_party_auth.pipeline.get', return_value=pipeline_partial):
+                    with patch('third_party_auth.pipeline.running', return_value=True):
+                        with patch('third_party_auth.is_enabled', return_value=True):
+                            reg.skip_email_verification = True
+                            inactive_user_view(request)
+                            self.assertEquals(user.is_active, True)
+                            assert not email.called, 'method should not have been called'
```

Comment:
we are using `self.assertEquals` and `assert`, better if we use either `self.assert*` methods or `assert` for both